webapp/src/app/shared/table/table.component.ts

```diff
@@ -36,10 +38,15 @@ export interface FilterItem {
     styleUrls: ['./table.component.css'],
 })
 export class TableComponent implements OnInit, AfterViewInit, OnChanges, OnDestroy {
+    @Input() idColumn;
+    @Input() selectedRowId;
     @Input() columnsAndFiltersToExcludeFromCasing = [];
     @Input() centeredColumns: { [key: string]: boolean } = {};
     @Input() columnsSortFunctionMap;
     @Input() filterFunctionMap;
+    @Input() dateCategoryList;
+    screenWidth: number;
+    whitelistColumnsSubscription: any;
     @Input() set columnWidths(value: { [key: string]: number }) {
         this._columnWidths = value;
         this.denominator = Object.keys(value).reduce((acc, next) => acc + value[next], 0);
```

Comment:
@sidharthjain-zemoso here and line#279: please, create a private method for updating the denominator